src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProjectFactory.cs

```diff
@@ -63,6 +77,7 @@ private VisualStudioProject CreateVisualStudioProject(string languageName, strin
 
             if (languageName == LanguageNames.FSharp)
             {
+                await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
                 var shell = (IVsShell)ServiceProvider.GlobalProvider.GetService(typeof(SVsShell));
```

Comment:
@jasonmalinowski can we use IAsyncServiceProvider here?  I'm not really sure i understand where/when/how we can use async stuff from teh platform itself.

Reply:
Yep, should be good to do that. You can also potentially use https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.shell.interop.ivsshell7.loadpackageasync?view=visualstudiosdk-2019 so the load itself is a bit friendlier to everyone involved.

Reply:
done.